source/agora/test/ValidatorRecurringEnrollment.d

```diff
@@ -174,3 +183,109 @@ unittest
     assertThrown!AssertError(createAndExpectNewBlock(
         Height(GenesisValidatorCycle)));
 }
+
+// Not all validators can enroll at the same height again. They should enroll
+// in 2 subsequent blocks. Nodes that can't enroll in the first block should
+// create another enrollment request for the next block
+unittest
+{
+    static class SocialDistancingNominator : TestNominator
+    {
+        public this (immutable(ConsensusParams) params, Clock clock,
+            NetworkManager network, KeyPair key_pair, Ledger ledger,
+            EnrollmentManager enroll_man, TaskManager taskman, string data_dir,
+            ulong txs_to_nominate, ulong test_start_time)
+        {
+            super(params, clock, network, key_pair, ledger, enroll_man, taskman, data_dir,
+            txs_to_nominate, test_start_time);
+        }
+
+        protected override bool prepareNominatingSet (out ConsensusData data) @safe
+        {
+            auto ret = super.prepareNominatingSet(data);
+            if (data.enrolls.length > 3)
+                data.enrolls.length = 3;
+            return ret;
+        }
+    }
+
+    static class SocialDistancingValidator : TestValidatorNode
+    {
+        public this (Config config, Registry* reg, immutable(Block)[] blocks,
+            in TestConf test_conf, shared(time_t)* cur_time)
+        {
+            super(config, reg, blocks, test_conf, cur_time);
+        }
+        ///
+        protected override TestNominator getNominator (Clock clock,
+            NetworkManager network, Ledger ledger, EnrollmentManager enroll_man,
+            TaskManager taskman)
+        {
+            return new SocialDistancingNominator(
+                this.params, clock, network, this.config.validator.key_pair,
+                ledger, enroll_man, taskman, this.config.node.data_dir,
+                this.txs_to_nominate, this.test_start_time);
+        }
+
+    }
+
+    static class SocialDistancingAPIManager : TestAPIManager
+    {
+        public this (immutable(Block)[] blocks, TestConf test_conf,
+            time_t initial_time)
+        {
+            super(blocks, test_conf, initial_time);
+        }
+
+        public override void createNewNode (Config conf, string file = __FILE__,
+            int line = __LINE__)
+        {
+            if (conf.validator.enabled)
+                this.addNewNode!SocialDistancingValidator(conf, file, line);
+            else
+                this.addNewNode!TestFullNode(conf, file, line);
+        }
+    }
+
+    TestConf conf = {
+        quorum_threshold : 100
+    };
+
+    auto network = makeTestNetwork!SocialDistancingAPIManager(conf);
+    network.start();
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.waitForDiscovery();
+
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Get the genesis block, make sure it's the only block externalized
+    auto blocks = node_1.getBlocksFrom(0, 2);
```

Comment:
Using 1 for the second parameter is more accurate.

Reply:
It's defensive programming, I like his approach. If you set the parameter to 1 then the next assert is pointless.